packages/core/src/components/search/Filter/FilterSlider.tsx

```diff
@@ -141,7 +143,14 @@ function FilterSlider({
   const isDeliveryPromiseEnabled = deliveryPromise.enabled
   const isPickupAllEnabled =
     deliverySettingsData?.deliveryMethods?.pickupAll?.enabled ?? false
-  const defaultPickupPoint = pickupPoints?.[0] ?? undefined
+
+  const selectedPickupPointId = state.selectedFacets.find(
+    ({ key }) => key === 'pickupPoint'
+  )?.value
+  const defaultPickupPoint =
+    pickupPoints?.find(({ id }) => id === selectedPickupPointId) ??
+    pickupPoints?.[0] ??
+    undefined
```

Comment:
suggestion: I believe we can change those namings because it's a bit confusing to call defaultPickupPoint equal to selectedPickupPoint, what do you think about this suggestion?
```suggestion
  const defaultPickupPoint = pickupPoints?.[0] ?? undefined
  const selectedPickupPointId = state.selectedFacets.find(
    ({ key }) => key === 'pickupPoint'
  )?.value
  const selectedPickupPoint =
    pickupPoints?.find(({ id }) => id === selectedPickupPointId) ??
    defaultPickupPoint
```

Reply:
hmm I think it's a good idea and it also differ the id from the object.